examples/vision/image_captioning.py

```diff
@@ -0,0 +1,486 @@
+"""
+Title: Image Captioning
+Author: [A_K_Nain](https://twitter.com/A_K_Nain)
+Date created: 2021/05/29
+Last modified: 2021/06/29
+Description: Implement image captioning model using CNN and Transformers.
+"""
+
+
+"""
+## Setup
+"""
+
+import os
+import re
+import numpy as np
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras.applications import efficientnet
+from keras.layers.experimental.preprocessing import TextVectorization
+
+
+"""
+## Download the dataset
+
+We will be using the Flickr8K dataset for this tutorial. This dataset comprises of more
+than 8,000 images that are each paired with five different captions.
+"""
+
+
+"""shell
+! wget -q https://github.com/jbrownlee/Datasets/releases/download/Flickr8k/Flickr8k_Dataset.zip
+! wget -q https://github.com/jbrownlee/Datasets/releases/download/Flickr8k/Flickr8k_text.zip
+! unzip -qq Flickr8k_Dataset.zip
+! unzip -qq Flickr8k_text.zip
+! rm Flickr8k_Dataset.zip Flickr8k_text.zip
+"""
+
+
+# Path to the images
+IMAGES_PATH = "Flicker8k_Dataset"
+
+# Desired image dimensions
+IMAGE_SIZE = (299, 299)
+
+# Vocabulary size
+VOCAB_SIZE = 10000
+# Maximum length allowed for any sequence
+MAX_SEQ_LENGTH = 40
```

Comment:
It's not a max in this case, it's a fixed length. There can be no shorter sequence (they get padded). `SEQ_LENGTH`.

Reply:
Ahh yes, I will fix this